286: Add a link to CSR in the PR's first comment

lgxbslgx opened the pull request:
Hi all,

This patch adds a link to the csr in the PR's first comment under **Issue**.
And the corresponding test is added.

Thanks for taking the time to review.

Best Regards,
-- Guoxiong

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-286](https://bugs.openjdk.java.net/browse/SKARA-286): Add a link to CSR in the PR's first comment


### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to c023b808f42b88c96cc90714467a701c8478c705
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1244/head:pull/1244` \
`$ git checkout pull/1244`

Update a local copy of the PR: \
`$ git checkout pull/1244` \
`$ git pull https://git.openjdk.java.net/skara pull/1244/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1244`

View PR using the GUI difftool: \
`$ git pr show -t 1244`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1244.diff">https://git.openjdk.java.net/skara/pull/1244.diff</a>

</details>

magicus reviewed (approved):
Looks good to me, but please await Erik's review as well.

erikj79 commented on bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:
```diff
@@ -116,11 +116,35 @@ private boolean isTargetBranchAllowed() {
+                var relationship = link.relationship();
+                if (relationship.isPresent() && relationship.get().equals("csr for")) {
+                    csr = link.issue().orElseThrow(
+                            () -> new IllegalStateException("Link with title 'csr for' does not contain issue")
```
I don't think we should throw an exception here. If we do, then a bad state in JBS will completely prevent the PullRequestBot from updating the status of the affected PR. I would prefer logging a warning and returning Optional.empty().

lgxbslgx commented:
@magicus Thanks for your review. I updated the code to resolve the null exception.

Also wait @erikj79 to review.

lgxbslgx commented:
Thanks for the review. I updated the code just now.

> I would suggest a bold CSR within normal parentheses after the title.

Added.

> I don't think we should throw an exception here

Fixed.

And I find the intergration blocker shouldn't invoke the method `CheckRun#issues` directly, so I add a parameter `withCsr` to the method `CheckRun#issues` to separate these two situations.

erikj79 reviewed (approved):
Looks good!

lgxbslgx commented:
@erikj79 @magicusThanks for the reviews. Could I get your help to sponsor this patch?

/integrate

lgxbslgx commented:
I updated the code to reduce the nested `if` statements and reduce the number of iterations in `for` statement.

lgxbslgx commented:
/integrate

erikj79 reviewed (approved):
I can sponsor this.

erikj79 commented:
/sponsor
